### PR TITLE
refactor(loader): v5.0.0

### PR DESCRIPTION
**Pull request by d3viant0ne**
##### Compatibility issues

- [x] None (  YAY !!!!!! ) 

##### Commit message body
```
 - Enforce options schema
 - Updates code style to Prettier
 - Updates engines
 - Updates peerDependencies
 - Enforces commit message formatting
 - Migrates to CircleCI 2.0

BREAKING CHANGE: Drops support for NodeJS 4.x

BREAKING CHANGE: Drops support for Webpack 2.x

BREAKING CHANGE: Less Loader now enforces it's options & types
 incorrect configurations with throw a validation error.

```

@jhnns - A note on the changes to the testing infrastructure. Non windows validation is now handled via CircleCI 2.0 on custom build containers.

The validation runs analysis & validations on the lowers common denominator ( maint LTS node + webpack latest ). If those to jobs pass, if will run the remaining validations in a `fan-out` configuration.

End result is a 50% reduction in build feedback, complete control over our build / test environment & validation on `webpack@next` as well as `node@next`.

There is a release task in the build setup as well that can be enabled at the request of the lib owner ( in this case you ).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 import processResult from './processResult';
 import getOptions from './getOptions';
+import schema from './options.json';
 
 const render = pify(less.render.bind(less));
 
 function lessLoader(source) {
   const loaderContext = this;
   const options = getOptions(loaderContext);
+  validateOptions(schema, options, 'lessLoader');
```
Comment (michael-ciniawsky) [inline src/index.js]: `\nvalidateOptions(...)\n` && `lessLoader` => `Less Loader`

File: src/formatLessError.js [hunk 2]
```diff
@@ -42,7 +42,8 @@ function getFileExcerptIfPossible(lessErr) {
  * @param {Array<string>} lessErr.extract - e.g. ['    .my-style {', '      .undefined-mixin;', '      display: block;']
  * @returns {LessError}
  */
-function formatLessError(err) { /* eslint-disable no-param-reassign */
+function formatLessError(err) {
```
Comment (michael-ciniawsky) [inline src/formatLessError.js]:
```js
class LessError extends Error {
   super();
   
   this.name = 'Syntax Error';
   this.message = `${err.message}\n\n`;
   
   // ...message formating

   Error.captureStackTrace(this, this.constructor)
}

export default LessError;
```

Comment (michael-ciniawsky) [reply src/formatLessError.js]: Maybe a separate PR aswell

**Comment by jhnns** [comment]
@d3viant0ne I guess this has been superseded by https://github.com/webpack-contrib/less-loader/pull/253? Anyway, thanks 👍

**Pull request closed without merging**